debezium-core/src/test/java/io/debezium/transforms/ExtractNewRecordStateTest.java

```diff
@@ -367,8 +382,8 @@ public void testAddSourceFields() {
 
             final SourceRecord createRecord = createComplexCreateRecord();
             final SourceRecord unwrapped = transform.apply(createRecord);
-            assertThat(((Struct) unwrapped.value()).get("__lsn")).isEqualTo(1234);
-            assertThat(((Struct) unwrapped.value()).getString("__version")).isEqualTo("version!");
+            assertThat(((Struct) unwrapped.value()).get("source__lsn")).isEqualTo(1234);
```

Comment:
How comes that this has changed? That'll be quite disruptive for existing users, and it's not something we should do. That said, it feels like we've cornered ourselves with the design of the existing options. In hindsight, it'd have been better to have something more generic like:

```
add.fields=op,ts_ms,source.txid,source.lsn
add.headers=op,...
```

But I can't see right now a good way for getting there without breaking existing consumers.

Reply:
So here's what I suggest:

* deprecate `add.source.fields` and `operation.header`
* add `add.fields` and `add.fields`; they take _simple_ names generally as values, e.g. txid, lsn, op etc.; column names will be __txid, __lsn, __op etc., as today. With the exception of `ts_ms` that's not ambiguous; in order to add source.ts_ms, the value source.ts_ms will have to be given (for the sake of regularity, that "qualified" form would work for all the other fields, too, but generally won't be needed to be used.). If a field is given in a qualified way, it's field name will be qualified, too, e.g. __source_ts_ms

Would that work?

Reply:
Thanks for your feedback @gunnarmorling. I agree that it should be more generic. I will implement it in the way you propose, with `add.fields` and `add.headers`.